src/App.jsx

```diff
@@ -1,65 +1,136 @@
-import React, { useRef, useState } from "react";
-import EosForm from "./EosForm";
+import React, { useRef, useState, useEffect } from "react";
+import ReactDOMServer from "react-dom/server";
+import * as EOSIcons from "eos-icons-react";
+// eslint-disable-next-line import/no-unresolved
+import { copyText } from "clipboard";
+import FormHolder from "./FormHolder";
+import EOSIconsList from "./converted-icons.json";
+import OptionsList from "./optionList.json";
 
 const App = () => {
-  const refContainer = useRef();
-  const [plugin, updatePlugin] = useState(
+  const searchCategory = useRef();
+  const searchTheme = useRef();
+  const inputField = useRef();
+  const [helperText, setHelperText] = useState(
     <p>Let&apos;s start by searching abstract.</p>
   );
-  const [file, updateFile] = useState(null);
-  const [searchGif, updateSearchGif] = useState(<div />);
+  const [iconsContainer, updateIcons] = useState(null);
+  const handleKeyUp = (event) => {
+    setHelperText(<p>We would be searching for {inputField.current.value}</p>);
+    if (event.key === "Enter") {
+      event.preventDefault();
+    }
+  };
 
-  const onSearch = () => {
-    const svgName = refContainer.current.value;
-    const request = new XMLHttpRequest();
-    const url = `https://cdn.jsdelivr.net/npm/eos-icons@latest/svg/${svgName}.svg`;
-    request.open("GET", url);
-    updateSearchGif(
-      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
-        <path
-          d="M12,2A10,10,0,1,0,22,12,10,10,0,0,0,12,2Zm0,18a8,8,0,1,1,8-8A8,8,0,0,1,12,20Z"
-          opacity=".5"
-        />
-        <path d="M20,12h2A10,10,0,0,0,12,2V4A8,8,0,0,1,20,12Z">
-          <animateTransform
-            attributeName="transform"
-            type="rotate"
-            from="0 12 12"
-            to="360 12 12"
-            dur="1s"
-            repeatCount="indefinite"
-          />
-        </path>
-      </svg>
+  const copyToClipboard = (svg) => {
+    copyText(ReactDOMServer.renderToStaticMarkup(svg));
+  };
+  const searchIconsByName = (name, theme, option) => {
+    const iconDivs = [];
+    // eslint-disable-next-line no-plusplus
+    for (let i = 0; i < EOSIconsList[option].length; i++) {
+      const icon = EOSIconsList[option][i];
+      const isFilledAvailable = "hasOutlined" in icon && icon.hasOutlined;
+      const isFilledSelected = theme === "Filled";
+      if (
+        icon.name.indexOf(name) !== -1 &&
+        (!isFilledSelected || (isFilledSelected && isFilledAvailable))
+      ) {
+        const nameIcon = `Eos${icon.name}${
+          isFilledSelected ? "Filled" : "Outlined"
+        }`;
+        const EOSReactIcon = EOSIcons[nameIcon];
+        if (EOSReactIcon === undefined) {
+          // eslint-disable-next-line no-continue
```

Comment:
just so this code is a bit cleaner, we can negate this comparison and move the next part of the script inside the `if`, so that we don't need the `continue` here and don't need to add exceptions to our coding conventions